google/cloud/bigquery_storage_v1/writer.py

```diff
@@ -316,64 +173,46 @@ def close(self, reason: Optional[Exception] = None):
                 an "intentional" shutdown. This is passed to the callbacks
                 specified via :meth:`add_close_callback`.
         """
-        self._shutdown(reason=reason)
-
-    def _shutdown(self, reason: Optional[Exception] = None):
-        """Run the actual shutdown sequence (stop the stream and all helper threads).
-
-        Args:
-            reason:
-                The reason to close the stream. If ``None``, this is
-                considered an "intentional" shutdown.
+        self._connection.close(reason=reason)
```

Comment:
read lock?

Reply:
Indeed it would be safer to add a lock to prevent the method from being called again.

Reply:
done